client/src/app/pages/applications/applications-table-analyze/applications-table-analyze.tsx

```diff
@@ -749,6 +757,13 @@ export const ApplicationsTableAnalyze: React.FC = () => {
           }}
         />
       )}
+
+      <SimpleDocumentViewerModal<Task | string>
```

Comment:
LGTM! 

I was wondering why you chose to pass the fetch functions down as props rather than use them inside the component itself. Seems like since we are already using conditional logic to determine the language type [here](https://github.com/konveyor/tackle2-ui/pull/1072/files#diff-4835af0eed4a8f71b4f6d20783d916be40890f398377685be99c7db990be71b9R79-R88), then we can just use the appropriate fetch functions within that conditional? Does that make sense? 

Either way, I like the use of typescript to make the component more extensible down the road. Maybe that was your intention. Just curious.

Reply:
Initially I wanted to have a more generic component that could be used elsewhere, and doing the fetch as a prop allows the location and methodology for the fetch can vary as needed.

In this PR, I eventually found 3 places that did a `window.open` or a Link with `target="_blank"`.  Two of those places used 1 URL, and the Link style was another.  So more generic FTW!  And playing with some generic typing with function type overloading got interesting.

One use:
https://github.com/konveyor/tackle2-ui/blob/aeeba069881d5bbb390b873199f7dd424029f654/client/src/app/pages/applications/components/application-detail-drawer/application-detail-drawer-analysis.tsx#L137-L142


The other use:
https://github.com/konveyor/tackle2-ui/blob/aeeba069881d5bbb390b873199f7dd424029f654/client/src/app/pages/applications/components/application-detail-drawer/application-detail-drawer-analysis.tsx#L113-L118